src/weights.jl

```diff
@@ -41,6 +46,7 @@ end
 Base.getindex(wv::W, ::Colon) where {W <: AbstractWeights} = W(copy(wv.values), sum(wv))
 
 @propagate_inbounds function Base.setindex!(wv::AbstractWeights, v::Real, i::Int)
+    isfinite(v) || throw(ArgumentError("weights cannot contain Inf or NaN values"))
     s = v - wv[i]
     wv.values[i] = v
     wv.sum += s
```

Comment:
technically `wv.sum` could become infinite here even if `v` is finite, so another check here might be added just for safety.

Reply:
ok makes sense, i swapped it to check the sum

Reply:
I meant to keep the other check also. The reason is that then we fail fast in a more common case without invalidating `wv`.

Reply:
Yeah good point to fail before invalidating wv. Is it worth having two checks for each new value? Could do something like below:
```suggestion
    s = v - wv[i]
    new_sum = wv.sum + s
    isfinite(new_sum) || throw(ArgumentError("weights cannot contain Inf or NaN values"))
    wv.values[i] = v
    wv.sum = new_sum
```

Reply:
right - this would be better